packages/malloy/src/lang/ast/expressions/expr-func.ts

```diff
@@ -127,23 +127,21 @@ export class ExprFunc extends ExpressionDef {
   ): ExprValue {
     const argExprsWithoutImplicit = this.args.map(arg => arg.getExpression(fs));
     if (this.isRaw) {
-      let collectType: CastType | undefined;
       const funcCall: SQLExprElement[] = [`${this.name}(`];
-      for (const expr of argExprsWithoutImplicit) {
-        if (collectType) {
+      argExprsWithoutImplicit.forEach((expr, i) => {
+        if (i !== 0) {
           funcCall.push(',');
-        } else {
-          if (isCastType(expr.type)) {
-            collectType = expr.type;
-          }
         }
         funcCall.push(expr.value);
-      }
+      });
       funcCall.push(')');
 
-      const dataType = this.rawType ?? collectType ?? 'number';
+      const inferredType = argExprsWithoutImplicit[0] ?? {type: 'number'};
```

Comment:
This is a little strange to set the result type to just the whole expression of the first argument... but the type system is happy with it because a `TypeDesc` _is_ a `ExpressionValueTypeDef`. @mtoy-googly-moogly if there were a way to extract only the `ExpressionValueTypeDef` parts of a `TypeDesc`, then I would use that here.